osu.Game/Online/Chat/Channel.cs

```diff
@@ -36,6 +36,11 @@ private long[] userIds
         /// </summary>
         public readonly SortedList<Message> Messages = new SortedList<Message>(Comparer<Message>.Default);
 
+        /// <summary>
+        /// Contains all the messages that weren't read by the user.
+        /// </summary>
+        public IEnumerable<Message> UnreadMessages => Messages.Where(m => LastReadId < m.Id);
```

Comment:
Just to be extra sure, is it documented anywhere that message IDs are temporally sequential (as in lower ID ⇔ earlier message)? Poking web sources it does seem to be the case but I'd like to be 100% on that.

Reply:
it will 100% be the case (primary key of table).